core/definitions/src/chain.ts

```diff
@@ -188,6 +186,28 @@ export abstract class ChainContext<
     return protocolIsRegistered(this.chain, protocolName);
   }
 
+  /**
+   * Construct a protocol client for the given protocol
+   *
+   * @param protocolName The name of the protocol to construct a client for
+   * @returns An instance of the protocol client that implements the protocol interface for the chain
+   */
+  async getProtocol<PN extends ProtocolName>(
+    protocolName: ProtocolName,
+    rpc?: RpcConnection<P>,
+  ): Promise<ProtocolImplementation<P, PN>> {
+    if (!this.protocols.has(protocolName)) {
+      const ctor = this.platform.getProtocolInitializer(protocolName);
+
+      const protocol = rpc
+        ? await this.platform.getProtocol(protocolName, rpc)
+        : new ctor(this.network, this.chain, await this.getRpc(), this.config.contracts);
```

Comment:
kinda confusing. would this be valid?

```suggestion
      const protocol = await this.platform.getProtocol(protocolName, rpc ?? await this.getRpc());
```

or are these two cases not equivalent for some reason?

Reply:
they're not equivalent, the first case will try to resolve the network/chain from the RPC, the second we pass it explicitly